schema/proto.go

```diff
@@ -201,54 +205,40 @@ func getTagStringFromTags(tags map[string]string) string {
 	return strings.Join(pairs, ", ")
 }
 
-func getMapTag(tp reflect.Type) (tags map[string]string, err error) {
+func getMapTag(typ reflect.Type) (tags map[string]string, err error) {
 	tags = make(map[string]string)
 	tags[ParquetType] = ParquetTypeMap
-	tags[ParquetConvertedType] = ParquetTypeMap
-	if IsPrimitiveOrPointerGoKind(tp.Key().Kind()) {
-		elementTags, elementErr := GetDefinedTypeTag(tp.Key())
-		if elementErr != nil {
-			return nil, elementErr
-		}
-		if elementType, exists := elementTags[ParquetType]; exists && IsPrimitiveParquetType(elementType) {
-			tags[ParquetKeyType] = elementType
-			if elementConvertedType, exists := elementTags[ParquetConvertedType]; exists {
-				tags[ParquetKeyConvertedType] = elementConvertedType
-			}
-		}
+	tags, err = updateKeyOrValueType(tags, typ, true)
+	if err != nil {
+		return nil, err
 	}
-	if IsPrimitiveOrPointerGoKind(tp.Elem().Kind()) {
-		elementTags, elementErr := GetDefinedTypeTag(tp.Elem())
+	return updateKeyOrValueType(tags, typ, false)
+}
+
+func updateKeyOrValueType(tags map[string]string, typ reflect.Type, updateKey bool) (map[string]string, error) {
+	if IsPrimitiveOrPointerGoKind(typ.Elem().Kind()) {
+		elementTags, elementErr := GetDefinedTypeTag(typ.Elem())
```

Comment:
Are you sure this is accurate now? Previously you had both `tp.Key().Kind()` and `tp.Elem().Kind()` in the logic and accordingly you captured their types via:

```go
...
elementTags, elementErr := GetDefinedTypeTag(tp.Key())
...
elementTags, elementErr := GetDefinedTypeTag(tp.Elem())
...
```

Now it seems like you're **only** doing this with `tp.Elem()`. Am I missing something?

Reply:
Oh Good catch. Need to fix it